### PR TITLE
feat(RegisterModel): 9914 autofill connection

### PR DESCRIPTION
**Pull request by gitdallas**
closes: https://issues.redhat.com/browse/RHOAIENG-9914

image
image
image
image
image
image
image

## Description
On the register modal page, there will be a new button to the right of Model Location / Object Storage (when selected) that says "Autofill from data connection".

Clicking that should bring up a modal with a project selector and a data connection selector. Selecting a project will populate the data connection selector with it's available data connections that contain a bucket.

Selecting a data connection and hitting the submit button will populate the Object Storage region/endpoint/bucket in the register model form.

## How Has This Been Tested?
Made sure that projects appear in project selector, and data connection is disabled with a note about selecting a project.
Upon selecting a project, data connection will either be populated or remain disabled with a note that the project does not contain any data connections.

Selected multiple data connections and made sure that the form populated with data from that data connection.

## Test Impact
Added tests to the register model to check that the button appears when it should, the modal appears when the button is clicked, the correct labeling/content appears in the dropdowns, and the fields get populated correctly.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/pages/modelRegistry/screens/RegisterModel/ConnectionModal.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            invalidDropdownPlaceholder="Select project"
+          />
+        </FormGroup>
+        <FormGroup label="Data connection name" isRequired fieldId="autofillConnection">
+          <ConnectionDropdown
+            onSelect={setConnection}
+            selectedConnection={connection}
+            project={project}
+          />
+          <HelperText>
```
Comment (mturley) [inline frontend/src/pages/modelRegistry/screens/RegisterModel/ConnectionModal.tsx:71]: We need to put `<FormHelperText>` around `<HelperText>` here for the correct spacing.

File: frontend/src/pages/modelRegistry/screens/RegisterModel/RegisterModel.scss [hunk 2]
```diff
@@ -0,0 +1,4 @@
+/* to help with getting the "Autofill" buttons to align properly */
+.pf-v5-c-radio__label {
+    width: 100%;
+}
```
Comment (mturley) [inline frontend/src/pages/modelRegistry/screens/RegisterModel/RegisterModel.scss:4]: We should avoid custom CSS, especially if it touches a PF class across the whole app. Maybe we can change this style rule so it only applies to this specific radio button, or look at an alternative... I'll get back to this in a sec

**Review by mturley — COMMENTED** [review]
A couple small things

File: frontend/src/pages/modelRegistry/screens/RegisterModel/ConnectionModal.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { Modal, Button, FormGroup, HelperText, Form } from '@patternfly/react-core';
+import ProjectSelector from '~/concepts/projects/ProjectSelector';
+import { DataConnection } from '~/pages/projects/types';
+import { ConnectionDropdown } from './ConnectionDropdown';
+
+export type ConnectionInfoType = {
```
[the remark this answers is not in the document]
Comment (gitdallas) [reply frontend/src/pages/modelRegistry/screens/RegisterModel/ConnectionModal.tsx:7]: must have been one of the things i backed out of 😆

**Pull request merged**